### PR TITLE
test: add unit tests for distributed/device_pool.py and strategies.py

### LINKED ISSUE #657: [Test]: Unit tests for distributed/device_pool.py and strategies.py
## Problem

`nightmarenet/distributed/device_pool.py` (1,557 bytes) and `nightmarenet/distributed/strategies.py` (1,853 bytes) are imported by `nightmarenet/training/trainer.py` (the core training loop) and `tests/test_distributed_native.py`. `strategies.py` also imports `DevicePool` internally. These modules manage GPU allocation and distributed training strategy selection (DP, DDP, FSDP) - incorrect behavior causes silent training failures, resource leaks, or crashes when switching between single/multi-GPU.

## Solution

Add unit tests:

**test_device_pool.py:**
- [ ] Pool creation with mocked CUDA devices
- [ ] Device allocation (acquire returns correct device index)
- [ ] Device release (returns device to pool)
- [ ] Pool exhaustion (all devices allocated, next request blocks or errors)
- [ ] Concurrent allocation safety (two threads requesting simultaneously)
- [ ] No-GPU fallback (CPU-only environment)

**test_distributed_strategies.py:**
- [ ] `apply_phase_strategy()` with DDP config
- [ ] `apply_phase_strategy()` with single-GPU (no wrapping)
- [ ] `apply_phase_strategy()` with CPU fallback
- [ ] Config parsing: valid strategy names, invalid strategy name error
- [ ] `unwrap_model()` correctly strips DDP/FSDP wrappers

## Acceptance Criteria

- `tests/test_device_pool.py` with 5+ tests
- `tests/test_distributed_strategies.py` with 5+ tests
- All tests run on CPU using mocked `torch.cuda` (monkeypatch `torch.cuda.device_count()`, `torch.cuda.is_available()`)
- Tests complete in under 10 seconds
- Does not duplicate `test_distributed_native.py` coverage

## Impact Radius

| File | Why it may need attention |
|------|--------------------------|
| `nightmarenet/distributed/device_pool.py` | Primary target |
| `nightmarenet/distributed/strategies.py` | Primary target |
| `nightmarenet/distributed/__init__.py` | Re-exports both modules |
| `nightmarenet/training/trainer.py` | Primary consumer (imports `DevicePool`, `apply_phase_strategy`) |
| `tests/test_distributed_native.py` | Existing tests - check coverage overlap |
| `nightmarenet/distributed/ddp_wrapper.py` | Related module that wraps models |

## Lint Scope

- Run `pytest tests/test_device_pool.py tests/test_distributed_strategies.py -v`
- Run `pytest tests/test_distributed_native.py -v` to confirm no regression
- Verify mocked `torch.cuda` doesn't leak into other test files

> **Important:** Please read CONTRIBUTING.md in full before requesting assignment. Assignment requests without a technical approach will not be considered. PRs demonstrating thorough research, benchmarks, or creative approaches beyond minimum criteria may earn bonus XP labels.

### PR DESCRIPTION
**Pull request by Babin123456**
## Summary
Introduces dedicated unit test suites `tests/test_device_pool.py` and `tests/test_distributed_strategies.py` to validate GPU discovery, device pool resource management, and phase-specific distributed strategies on CPU without requiring physical CUDA hardware.

## Related Issues
Closes #657

## Motivation
`nightmarenet/distributed/device_pool.py` and `nightmarenet/distributed/strategies.py` orchestrate multi-device execution and determine strategy wrappers (DDP, DataParallel, Single-Device) across training phases. Dedicated unit test coverage ensures device pooling logic, CPU fallbacks, memory calculations, and recursive model unwrapping execute reliably in all environments.

## Changes Made
- Created `tests/test_device_pool.py` with 7 unit tests:
  - Auto-discovery with mocked CUDA devices (`torch.cuda.device_count=4`, `torch.cuda.is_available=True`).
  - Explicit override device list (`override_devices=[1, 3]`).
  - Empty override list and single-GPU setups (`should_use_ddp` evaluation).
  - CPU-only fallback when CUDA is unavailable.
  - VRAM memory estimation formula verification (`estimate_memory_requirements`).
  - Multi-threaded concurrent inspection safety.
- Created `tests/test_distributed_strategies.py` with 10 unit tests:
  - `unwrap_model`: Direct unwrapped models, single-layer wrappers, and deeply nested recursive wrappers.
  - `apply_phase_strategy`: Disabled distributed execution, single-device fallback, wake & nightmare DDP wrapping, uninitialized DDP fallback, dream phase `DataParallel` wrapping with available device IDs, compress phase single-device enforcement, and unknown phase fallbacks.
- Synchronized `docs/api/openapi.json` to prevent OpenAPI drift.

## Testing & Verification
- Ran all unit tests on CPU:
  ```bash
  uv run pytest tests/test_device_pool.py tests/test_distributed_strategies.py

### CODE REVIEW

**Comment by Babin123456** [comment]
@Adit-Jain-srm

**Review by Adit-Jain-srm — APPROVED** [review]
